src/rpc/fts/analyzer/FourthOrderParameter.tpp

```diff
@@ -126,6 +126,11 @@ namespace Rpc {
    template <int D>
    void FourthOrderParameter<D>::computeFourthOrderParameter()
    {
```

Comment:
Perhaps add a precondition check

UTIL_CHECK(system().w().hasData());

to this and other analyzers. 

This is a trivial test, that will probably always pass when the code is not badly broken. Nonetheless, it's a good habit to write preconditions that systematically check everything that needs to be true on entry to a function - the code does a much better job of reporting a meaningful error when subsequent changes break things if thorough precondition tests are in place.